### PR TITLE
Wait for webhook readiness and healthy 

### PR DESCRIPTION
**Pull request by geetikabatra**
This change adds a ready and health check on the webhook server.

Release note:

Signed-off-by: Geetika Batra <geetika.b@vmware.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Add readiness and health checks to webhook server
```

### CODE REVIEW

**Comment by randomvariable** [comment]
@geetikabatra looks like you haven't signed the CNCF CLA?

/assign @sedefsavas 
as the last person who was touching this.

**Comment by geetikabatra** [comment]
@randomvariable, I had the wrong email set in my config. If @sedefsavas wants to take it forward. I will close this PR.

File: main.go [hunk 1]
```diff
@@ -261,12 +261,12 @@ func main() {
...
 	// +kubebuilder:scaffold:builder
 
-	if err := mgr.AddReadyzCheck("ping", healthz.Ping); err != nil {
+	if err := mgr.AddReadyzCheck("webhook", mgr.GetWebhookServer().StartedChecker()); err != nil {
 		setupLog.Error(err, "unable to create ready check")
 		os.Exit(1)
 	}
 
-	if err := mgr.AddHealthzCheck("ping", healthz.Ping); err != nil {
+	if err := mgr.AddReadyzCheck("webhook", mgr.GetWebhookServer().StartedChecker()); err != nil {
```
Comment (sbueringer) [inline main.go:269]:
```suggestion
	if err := mgr.AddHealthzCheck("webhook", mgr.GetWebhookServer().StartedChecker()); err != nil {
```

Comment (geetikabatra) [reply main.go:269]: @sbueringer Thanks for the review. But unfortunately closing it as @sedefsavas was already working on it.

**Comment by randomvariable** [comment]
@geetikabatra i meant @sedefsavas to look at the PR as we did previously have this and took it out and I wasn't directly involved in that decision.

**Comment by geetikabatra** [comment]
@randomvariable ohh okay! I will fix it as per @sbueringer 's comments.

File: main.go (hunk 1 above)
Comment (geetikabatra) [reply main.go]: @sbueringer Updated the suggested change. Please take a look. Thanks in advance.

Comment (sbueringer) [reply main.go]: @geetikabatra Looks good thx. I think you have to change your commit message: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2685#issuecomment-902636137 (and drop the link to the issue)

**Comment by sedefsavas** [comment]
@geetikabatra We should do the same for EKS. See: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2603

**Comment by geetikabatra** [comment]
@sedefsavas EKS has been moved out of the experimental feature. So, the change in EKS is not necessary now, I think.
I didn't see any other instance of ping being used anywhere else in the repository.

**Comment by sedefsavas** [comment]
> @sedefsavas EKS has been moved out of the experimental feature. So, the change in EKS is not necessary now, I think.
> I didn't see any other instance of ping being used anywhere else in the repository. I think we are good to go here. WDYT?

Right! Forgot about it. 

/ok-to-test

**Comment by randomvariable** [comment]
@geetikabatra 
Please remove

```
Fixes: #2602

Release note:
```

from the commit message. In general, Prow will fail commit messages that do `Fixes #` etc...

In addition, please review the errors:

`./main.go:39:2: imported and not used: "sigs.k8s.io/controller-runtime/pkg/healthz"`

File: main.go (hunk 1 above)
Comment (geetikabatra) [reply main.go]: Done!

**Comment by geetikabatra** [comment]
Done!

**Comment by sbueringer** [comment]
Thx!
/lgtm

**Pull request merged**